### PR TITLE
random read from guest long time

### PR DESCRIPTION
**Pull request by suqinhuang**
Read data from /dev/random 10 hours
Signed-off-by: Suqin Huang shuang@redhat.com
id: 1377564

### CODE REVIEW

**Comment by suqinhuang** [comment]
@pingl2016 could you help review this patch

File: qemu/tests/cfg/rng_read_longtime.cfg [hunk 1]
```diff
@@ -0,0 +1,20 @@
+- rng_read_longtime:
+    type = rng_read_longtime
+    kill_vm = yes
+    virt_test_type = qemu
+    test_duration = 36000
+    sub_test = rng_bat
+    no no_virtio_rng
+    Windows:
+        # Please update path of rng_dll_register_cmd to right path which included you driver
```
Comment (PandaWei) [inline qemu/tests/cfg/rng_read_longtime.cfg]: your driver?

Comment (PandaWei) [inline qemu/tests/cfg/rng_read_longtime.cfg]: your driver?

File: qemu/tests/cfg/rng_read_longtime.cfg [hunk 2]
```diff
@@ -0,0 +1,20 @@
+- rng_read_longtime:
+    type = rng_read_longtime
+    kill_vm = yes
+    virt_test_type = qemu
+    test_duration = 36000
```
Comment (PandaWei) [inline qemu/tests/cfg/rng_read_longtime.cfg]: which default value is better, 36000 or 60?
There's a large gap between them.

FYI, params.get("test_duration", "60")

File: qemu/tests/rng_read_longtime.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+from virttest import utils_test
+
+
+def run(test, params, env):
+    """
+    1) boot guest with virtio-rng device
+    2) check host random device opened by qemu (optional)
+    3) enable driver verifier in guest
+    4) read random data from guest
+    5) repeat 2~4 10 hours (can be set)
```
Comment (pingl2016) [inline qemu/tests/rng_read_longtime.py]: It maybe test step 2~4 during 10 hours.

Comment (suqinhuang) [inline qemu/tests/rng_read_longtime.py]: repeat step2~step4 10 hours, have been updated, thanks for review

File: qemu/tests/cfg/rng_read_longtime.cfg (hunk 1 above)
Comment (suqinhuang) [inline qemu/tests/cfg/rng_read_longtime.cfg]: it works with default setting currently.

File: qemu/tests/cfg/rng_read_longtime.cfg (hunk 2 above)
Comment (suqinhuang) [inline qemu/tests/cfg/rng_read_longtime.cfg]: update default value to "3600"

**Comment by weliao** [comment]
@suqinhuang 
2016-10-09 14:23:52,594 stacktrace       L0044 ERROR| OSError: [Errno 24] Too many open files
2016-10-09 14:23:52,594 stacktrace       L0045 ERROR|
2016-10-09 14:23:52,594 output           L0642 DEBUG| Exception AttributeError: "'ShellSession' object has no attribute 'reader_fds'" in <bound method ShellSession.__del__ of <aexpect.client.ShellSession object at 0x7fb0cc059250>> ignored
2016-10-09 14:23:52,594 test             L0572 ERROR| Traceback (most recent call last):

2016-10-09 14:23:52,594 test             L0572 ERROR|   File "/usr/lib/python2.7/site-packages/avocado_plugins_vt-41.0-py2.7.egg/avocado_vt/test.py", line 434, in _run_avocado
    raise test_exception

2016-10-09 14:23:52,594 test             L0572 ERROR| OSError: [Errno 24] Too many open files

2016-10-09 14:23:52,595 test             L0589 ERROR| ERROR 2-smp_4.4096m.repeat1.run_test.Host_RHEL.m7.u3.virtio_rng.rng_random.qcow2.virtio_scsi.up.virtio_net.RHEL.7.3.x86_64.io-github-autotest-qemu.rng_read_longtime -> OSError: [Errno 24] Too many open files

**Comment by suqinhuang** [comment]
@weliao hi, did you install your host manually or with profile

**Comment by weliao** [comment]
@suqinhuang   manually install, if because of this , I will try again use profile install host.

**Comment by pingl2016** [comment]
@suqinhuang @weliao I run the case on the host installed with profile. The issue mentioned by weliao cannot be reproduced, but hit the below issue after running the case for 11987 seconds with linux guest or running the case for 3173 seconds with windows guest.

09:18:28 ERROR| Traceback (most recent call last):
09:18:28 ERROR|   File "/usr/lib/python2.7/site-packages/avocado_plugins_vt-41.0-py2.7.egg/avocado_vt/test.py", line 196, in runTest
09:18:28 ERROR|     self._runTest()
09:18:28 ERROR|   File "/usr/lib/python2.7/site-packages/avocado_plugins_vt-41.0-py2.7.egg/avocado_vt/test.py", line 337, in _runTest
09:18:28 ERROR|     run_func(self, params, env)
09:18:28 ERROR|   File "/usr/share/avocado/data/avocado-vt/test-providers.d/downloads/io-github-autotest-qemu/qemu/tests/rng_read_longtime.py", line 23, in run
09:18:28 ERROR|     utils_test.run_virt_sub_test(test, params, env, sub_test)
09:18:28 ERROR|   File "/usr/lib/python2.7/site-packages/avocado_plugins_vt-41.0-py2.7.egg/virttest/utils_test/__init__.py", line 1242, in run_virt_sub_test
09:18:28 ERROR|     run_func(test, params, env)
09:18:28 ERROR|   File "/usr/lib/python2.7/site-packages/avocado_plugins_vt-41.0-py2.7.egg/virttest/error_context.py", line 135, in new_fn
09:18:28 ERROR|     return fn(*args, **kwargs)
09:18:28 ERROR|   File "/usr/share/avocado/data/avocado-vt/test-providers.d/downloads/io-github-autotest-qemu/qemu/tests/rng_bat.py", line 106, in run
09:18:28 ERROR|     session = vm.wait_for_login(timeout=timeout)
09:18:28 ERROR|   File "/usr/lib/python2.7/site-packages/avocado_plugins_vt-41.0-py2.7.egg/virttest/virt_vm.py", line 1042, in wait_for_login
09:18:28 ERROR|     username, password)
09:18:28 ERROR|   File "/usr/lib/python2.7/site-packages/avocado_plugins_vt-41.0-py2.7.egg/virttest/error_context.py", line 135, in new_fn
09:18:28 ERROR|     return fn(*args, **kwargs)
09:18:28 ERROR|   File "/usr/lib/python2.7/site-packages/avocado_plugins_vt-41.0-py2.7.egg/virttest/virt_vm.py", line 956, in login
09:18:28 ERROR|     interface=neigh_attach_if)
09:18:28 ERROR|   File "/usr/lib/python2.7/site-packages/avocado_plugins_vt-41.0-py2.7.egg/virttest/remote.py", line 254, in remote_login
09:18:28 ERROR|     handle_prompts(session, username, password, prompt, timeout)
09:18:28 ERROR|   File "/usr/lib/python2.7/site-packages/avocado_plugins_vt-41.0-py2.7.egg/virttest/remote.py", line 141, in handle_prompts
09:18:28 ERROR|     timeout=timeout, internal_timeout=0.5)
09:18:28 ERROR|   File "/usr/lib/python2.7/site-packages/aexpect/client.py", line 853, in read_until_last_line_matches
09:18:28 ERROR|     print_func)
09:18:28 ERROR|   File "/usr/lib/python2.7/site-packages/aexpect/client.py", line 767, in read_until_output_matches
09:18:28 ERROR|     max(0, end_time - time.time()))
09:18:28 ERROR| ValueError: filedescriptor out of range in select()
09:18:28 ERROR|
09:18:28 ERROR| Traceback (most recent call last):

09:18:28 ERROR|   File "/usr/lib/python2.7/site-packages/avocado_plugins_vt-41.0-py2.7.egg/avocado_vt/test.py", line 434, in _run_avocado
    raise test_exception

09:18:28 ERROR| ValueError: filedescriptor out of range in select()

09:18:28 ERROR| ERROR 1-smp_8.8192m.repeat1.run_test.Host_RHEL.m7.u3.virtio_rng.rng_random.qcow2.virtio_scsi.up.virtio_net.RHEL.7.3.x86_64.io-github-autotest-qemu.rng_read_longtime -> ValueError: filedescriptor out of range in select()

**Pull request closed without merging**